src/CosmoShips.sol

```diff
@@ -11,20 +11,29 @@ import "./IAttributeVerifier.sol";
 contract CosmoShips is ERC721, AccessControl, ReentrancyGuard, AttributeEncoder, ERC721Enumerable {
     bytes32 public constant MAINTENANCE_ROLE = keccak256("MAINTENANCE_ROLE");
     bytes32 public merkleRoot;
+    uint256 private _shipIdToMint;
     uint256 public mintPrice;
     IAttributeVerifier public verifier;
     mapping(uint256 => bool) public tokenMinted;
     mapping(uint256 => uint256) public attributes;
 
     event Minted(address minter, uint256 tokenId);
 
-    constructor(bytes32 _merkleRoot, uint256 _mintPrice, address _defaultAdmin, address _verifier)
-        ERC721("CosmoShips", "CSSS")
-    {
+    constructor(
+        bytes32 _merkleRoot,
+        uint256 _initialShipIdToMint,
```

Comment:
`_initialShipIdToMint` is redundant

Reply:
The reason I added it here is to be able to redeploy the script by using another initial value. Let's say for example that 10 ships have already been minted and we find a bug, and we need to redeploy. By not having that, we won't have control on what was the last id, and where the id need to be started. It will start from 1.

Reply:
https://docs.openzeppelin.com/contracts/4.x/api/utils#Counters

Reply:
redeploying and preserving the previous tokens are not possible.
We should be First Time Right (FTR)
In case of an upgrade, a burn-mint or some wrapper is required, and to a new contract.

Reply:
How it is not possible? Having the merkle root, and using the maintenance function, can't put the contract in the state it was before?